File: server/typescript/packages/render/test/email-document.test.ts
import { describe, test, expect } from "bun:test";
import type { EmailDocument } from "../src/index.js";

describe("EmailDocument", () => {
  test("constructs with all fields", () => {
    const doc: EmailDocument = {
      subject: "Hello",
      htmlBody: "<p>Hi</p>",
      textBody: "Hi",
    };
    expect(doc.subject).toBe("Hello");
    expect(doc.htmlBody).toBe("<p>Hi</p>");
    expect(doc.textBody).toBe("Hi");
  });

  test("textBody is optional (plain-text alternative omitted)", () => {
    const doc: EmailDocument = { subject: "S", htmlBody: "<b>H</b>" };
    expect(doc.subject).toBe("S");
    expect(doc.htmlBody).toBe("<b>H</b>");
    expect(doc.textBody).toBeUndefined();
  });
});

File: server/java/render/src/test/java/com/metaobjects/render/EmailDocumentTest.java
package com.metaobjects.render;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

public class EmailDocumentTest {

    @Test public void accessorsExposeAllFields() {
        EmailDocument doc = new EmailDocument("Hello", "<p>Hi</p>", "Hi");
        assertEquals("Hello", doc.subject());
        assertEquals("<p>Hi</p>", doc.htmlBody());
        assertEquals("Hi", doc.textBody());
    }

    @Test public void textBodyMayBeNull() {
        EmailDocument doc = new EmailDocument("S", "<b>H</b>", null);
        assertEquals("S", doc.subject());
        assertEquals("<b>H</b>", doc.htmlBody());
        assertNull(doc.textBody());
    }
}
